src/utils/phoenix-utils.js

```diff
@@ -141,3 +145,35 @@ export function getPresenceProfileForSession(presences, sessionId) {
   const presence = entry[1];
   return (presence && presence.metas && presence.metas[0].profile) || {};
 }
+
+// Takes the given channel, and creates a new channel with the same bindings
+// with the given socket, joins it, and leaves the old channel after joining.
+export function migrateChannelToSocket(oldChannel, socket, params) {
```

Comment:
Oh boy. It looks like we're reaching into phoenix library internals here? Dunno if it's worth adding a comment to warn our future selves to watch out for breakages in new versions of the library (though I guess we use our own fork at the moment)

Reply:
Yep I can add a comment. This seemed to be the better way to do it than making it so we can re-materialize all of the event handlers in `hub.js` into a new channel. I can add a comment. Also, i forgot we were on a fork, we can move off of it at some point: my changes were merged into master 😎